pkg/codec/modifier_base.go

```diff
@@ -242,16 +252,27 @@ func doForMapElements[T any](valueMapping map[string]any, fields map[string]T, f
 		path := strings.Split(key, ".")
 		name := path[len(path)-1]
 		path = path[:len(path)-1]
+
 		extractMaps, err := getMapsFromPath(valueMapping, path)
 		if err != nil {
-			return err
+			return PathMappingError{Err: err, Path: key}
```

Comment:
This no longer returns a common error so it won't unwrap when sent over a LOOP

Reply:
Ah ok. If I wrap the type error within `Error()` would that unwrap over LOOP?

Reply:
I think you can just add a `Cause() error` method so it can match the wrapped error

Reply:
@nolag Does this behave the way that we want? Or are the outer wrapped errors lost? (if so, I think we could change that easily)

Reply:
I think we are good:
https://github.com/grpc/grpc-go/blob/v1.64.0/status/status.go#L123
Original `err` message is used even after doing the `As` conversion, so outer wrapped message will pass through.